decimal64math.go

```diff
@@ -27,21 +29,29 @@ func (d Decimal64) Add(e Decimal64) Decimal64 {
 	if significand1 == 0 && significand2 == 0 {
 		return Zero64
 	}
-
 	exp1, significand1, exp2, significand2 = matchScales(exp1, significand1, exp2, significand2)
-
 	if sign1 == sign2 {
-		significand := significand1 + significand2
+		significand = significand1 + significand2
 		exp1, significand = renormalize(exp1, significand)
 		if significand > maxSig || exp1 > expMax {
 			return infinities[sign1]
 		}
 		return newFromParts(sign1, exp1, significand)
 	}
 	if significand1 > significand2 {
-		return newFromParts(sign1, exp1, significand1-significand2)
+		significand = significand1 - significand2
+		exp = exp1
+		sign = sign1
+	} else {
+		significand = significand2 - significand1
+		exp = exp2
+		sign = sign2
+	}
+	if significand == 0 {
+		return zeroes[sign]
 	}
-	return newFromParts(sign2, exp2, significand2-significand1)
+	exp, significand = renormalize(exp, significand)
```

Comment:
This means that all deicmal64s that represent the same "number" will have the same bit representation

This seemed faster than adding a clause in the Cmp function to match the numbers scales but will mean external number representations will need to be normalised to be used in our library